README.md

```diff
@@ -343,9 +343,32 @@ foreach ($annotation->faces() as $key => $face) {
 }
 ```
 
+## Caching Access Tokens
+
+By default the library will use a simple in-memory caching implementation, however it is possible to override this behavior by passing a [PSR-6](http://www.php-fig.org/psr/psr-6/) caching implementation in to the desired client.
+
+The following example takes advantage of [Symfony's Cache Component](https://github.com/symfony/cache).
+
+```php
+<?php
+require 'vendor/autoload.php';
+
+use Google\Cloud\Storage\StorageClient;
+use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+
+$cacheDir = '/tmp/google-cloud-cache';
+// Make sure to set the proper permissions to restrict access to tokens.
+@mkdir($cacheDir, 0600);
```

Comment:
0600 doesn't work. 0700 does work, but it has a security issue. Using the same directory among all the users leads to a serious secret leak.

1. User A use this library with their service account secret key A
2. Then the token is stored in a filesystem cache
3. User root use this library with other service account secret key B
4. Then the same directory is used for the cache, so the owner of the directory stays the same
5. The token is stored in the same filesystem cache
6. User A can read the token of the secret key B

To mitigate this, maybe you can use `${HOME}` for base dir, `${HOME}/.cache/google-cloud` or something. Strictly saying it is better check the permission of the directory in the step 4 above.